lib/server-jobs.js

```diff
@@ -208,10 +208,21 @@ module.exports.init = function () {
   }, config.serverJobHeartbeatIntervalSec * 1000);
 };
 
-module.exports.close = function () {
-  if (intervalId) {
-    clearInterval(intervalId);
-    intervalId = null;
+module.exports.stop = async function () {
+  await new Promise((resolve) => {
+    // Poll until all jobs are finished.
+    const pollIntervalId = setInterval(() => {
+      if (Object.keys(module.exports.liveJobs).length === 0) {
+        clearInterval(pollIntervalId);
+        resolve();
+      }
+    }, 100);
+  });
```

Comment:
Similar to the above, can we do this with a while loop and sleeps?

Reply:
Yep, done in 120a0c9.